network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -3461,11 +3693,11 @@ private <T extends ReactiveLimitHolder> void insertReactiveCapabilityCurvePointI
             equipment.setReactiveLimits(new ReactiveCapabilityCurveAttributes());
         }
         ReactiveLimitsAttributes reactiveLimitsAttributes = equipment.getReactiveLimits();
-        if (reactiveLimitsAttributes instanceof ReactiveCapabilityCurveAttributes) {
-            if (((ReactiveCapabilityCurveAttributes) reactiveLimitsAttributes).getPoints() == null) {
-                ((ReactiveCapabilityCurveAttributes) reactiveLimitsAttributes).setPoints(new TreeMap<>());
+        if (reactiveLimitsAttributes instanceof ReactiveCapabilityCurveAttributes reactiveCapabilityCurveAttributes) {
+            if (reactiveCapabilityCurveAttributes.getPoints() == null) {
+                reactiveCapabilityCurveAttributes.setPoints(new TreeMap<>(COMPARATOR));
```

Comment:
Do we keep (COMPARATOR) as the COMPARATOR is reinjected in the setter now ?

Reply:
better be sure ?

Reply:
Yes I think we can rely on network-store implementation to always inject thje right comparator